src/rust/bitbox02-rust/src/hww/api/backup.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::pb;
+use super::Error;
+
+use pb::response::Response;
+
+use crate::workflow::{confirm, status};
+use bitbox02::backup;
+
+pub async fn check(
+    &pb::CheckBackupRequest { silent }: &pb::CheckBackupRequest,
+) -> Result<Response, Error> {
+    if !bitbox02::sdcard_inserted() {
+        return Err(Error::COMMANDER_ERR_INVALID_INPUT);
+    }
+    match backup::check() {
+        Ok(backup::CheckData { id, name, .. }) => {
+            if !silent {
+                let params = confirm::Params {
+                    title: "Name?",
+                    body: &name,
+                    scrollable: true,
+                    ..Default::default()
+                };
+
+                if !confirm::confirm(&params).await {
+                    return Err(Error::COMMANDER_ERR_GENERIC);
```

Comment:
Since you're here, why not change it to `COMMANDER_ERR_USER_ABORT`? https://github.com/digitalbitbox/bitbox02-firmware/blob/75470b6/src/workflow/backup.c#L139

> // TODO Change to error "User Abort"

Reply:
It would technically be a breaking change and a major version bump, but if you agree, i'll just do the change and not bump the major version :sweat_smile: only the bitboxapp (api-go) and send_message use this api call currently, and the app will work the same without any change.

Reply:
yeah, agree. there isn't any reasonable benefit in bumping the major version because of this.

Reply:
don't forget there are two places.

Reply:
Done

Reply:
:heavy_check_mark: